internal/pkg/agent/application/upgrade/artifact/download/verifier_test.go

```diff
@@ -192,6 +194,117 @@ func TestVerifySHA512HashWithCleanup_failure(t *testing.T) {
 	}
 }
 
+func TestVerifySHA512HashWithCleanup_BrokenHashFile(t *testing.T) {
+
+	const data = "" +
+		"Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. " +
+		"Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. " +
+		"Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. " +
+		"Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum."
+	// if you change data, the constant below should be updated
+	const correct_data_hash = "8ba760cac29cb2b2ce66858ead169174057aa1298ccd581514e6db6dee3285280ee6e3a54c9319071dc8165ff061d77783100d449c937ff1fb4cd1bb516a69b9"
+
+	const filename = "lorem_ipsum.txt"
+	const hashFileName = filename + ".sha512"
+
+	type skipFunc func(t *testing.T)
+
+	type testcase struct {
+		name            string
+		skip            skipFunc
+		hash            []byte
+		hashPermissions fs.FileMode
+		wantErr         assert.ErrorAssertionFunc
+		wantLogSnippets []string
+	}
+
+	testcases := []testcase{
+		{
+			name:            "happy path - correct hash and format",
+			hash:            []byte(correct_data_hash + "  " + filename),
+			hashPermissions: 0o640,
+			wantErr:         assert.NoError,
+		},
+		{
+			name:            "happy path - broken lines before correct hash and format",
+			hash:            []byte("this_is just_filler\n" + "some_more_filler\n" + correct_data_hash + "  " + filename),
+			hashPermissions: 0o640,
+			wantErr:         assert.NoError,
+		},
+		{
+			name:            "truncated hash line - no filename",
+			hash:            []byte(correct_data_hash),
+			hashPermissions: 0o640,
+			wantErr:         assert.Error,
+			wantLogSnippets: []string{`contents: "` + correct_data_hash + `"`},
+		},
+		{
+			name:            "truncated hash",
+			hash:            []byte(correct_data_hash[:8] + "  " + filename),
+			hashPermissions: 0o640,
+			wantErr: func(t assert.TestingT, err error, i ...interface{}) bool {
+				target := new(ChecksumMismatchError)
+				return assert.ErrorAs(t, err, &target, "mismatched hash has a specific error type", i)
+			},
+		},
+		{
+			name:            "empty hash file",
+			hash:            []byte{},
+			hashPermissions: 0o640,
+			wantErr:         assert.Error,
+			wantLogSnippets: []string{`contents: ""`},
+		},
+		{
+			name: "non-existing hash file",
+			hash: nil,
+			wantErr: func(t assert.TestingT, err error, i ...interface{}) bool {
+				return assert.ErrorIs(t, err, fs.ErrNotExist, i)
+			},
+		},
+		{
+			name: "unreadable hash file",
+			skip: func(t *testing.T) {
+				if runtime.GOOS == "windows" {
+					t.Skip("permissions are not really testable on windows")
```

Comment:
I think we should explain better why it's not testable.

Reply:
reworded in a6f90fca54178566fc9d36600004f02a9bf90c30